server/plugin.go

```diff
@@ -280,15 +282,76 @@ func (p *Plugin) deleteGitlabUserToken(userID string) error {
 func (p *Plugin) getGitlabUserInfoByMattermostID(userID string) (*gitlab.UserInfo, *APIErrorResponse) {
 	var userInfo gitlab.UserInfo
 
-	if infoBytes, err := p.API.KVGet(userID + GitlabUserInfoKey); err != nil || infoBytes == nil {
-		return nil, &APIErrorResponse{ID: APIErrorIDNotConnected, Message: "Must connect user account to GitLab first.", StatusCode: http.StatusBadRequest}
+	infoBytes, err := p.API.KVGet(userID + GitlabUserInfoKey)
+
+	if err != nil || infoBytes == nil {
+		gitlabTokenBytes, err := p.API.KVGet(userID + GitlabMigrationTokenKey)
+		if err != nil || gitlabTokenBytes == nil {
+			return nil, &APIErrorResponse{ID: APIErrorIDNotConnected, Message: "Must connect user account to GitLab first.", StatusCode: http.StatusBadRequest}
+		} else {
+			return p.migrateGitlabToken(userID)
+		}
 	} else if err := json.Unmarshal(infoBytes, &userInfo); err != nil {
 		return nil, &APIErrorResponse{ID: "", Message: "Unable to parse user info.", StatusCode: http.StatusInternalServerError}
 	}
 
 	return &userInfo, nil
 }
 
+func (p *Plugin) migrateGitlabToken(userID string) (*gitlab.UserInfo, *APIErrorResponse) {
+	config := p.getConfiguration()
+
+	var userInfo struct {
+		gitlab.UserInfo
+		Token *oauth2.Token
+	}
```

Comment:
Just so I understand correctly, we need to declare this local struct, because we no longer use that token from this struct elsewhere, so we have removed it from the "known" struct definition in the KV store.